tests/Unit/DependencyInjection/FOSHttpCacheExtensionTest.php

```diff
@@ -662,10 +665,62 @@ public function testVarnishCustomTagsHeader()
         $this->assertEquals(['tag_mode' => 'ban', 'tags_header' => 'myheader'], $container->getParameter('fos_http_cache.proxy_client.varnish.options'));
     }
 
+    /**
+     * @param string|array|mixed $serversValue           string and array are ok, anything else not, string only if it uses an env var
+     * @param string|mixed|null  $envValue               should be a string, otherwise an error will show up
+     * @param array|null         $expectedServersValue
+     * @param string|null        $expectExceptionClass
+     * @param string|null        $expectExceptionMessage
+     *
+     * @dataProvider dataVarnishServersConfig
+     */
+    public function testVarnishServersConfig($serversValue, $serversFromJsonEnvValue, $envValue, $expectedServersValue, $expectExceptionClass, $expectExceptionMessage)
+    {
+        if ($expectExceptionClass) {
+            $this->expectException($expectExceptionClass);
+            $this->expectExceptionMessage($expectExceptionMessage);
+        }
+
+        $_ENV['VARNISH_SERVERS'] = $envValue;
+        $container = $this->createContainer();
+
+        // workaround to get the possible env string into the EnvPlaceholderParamterBag
+        $container->setParameter('triggerServersValue', $serversValue);
+        $container->setParameter('triggerServersFromJsonEnvValue', $serversFromJsonEnvValue);
+        (new ResolveParameterPlaceHoldersPass())->process($container);
+
+        $config = $this->getBaseConfig();
+        $config['proxy_client']['varnish']['http']['servers'] = $container->getParameter('triggerServersValue');
```

Comment:
we should only configure the option if we are supposed to configure it. then the `requiresAtLeastOneElement` works
```suggestion
        if (null !== $serversValue) {
            $config['proxy_client']['varnish']['http']['servers'] = $container->getParameter('triggerServersValue');
        }
```

Reply:
thanks - in addition, I need to unset the servers in the else case, as it is defined in `getBaseConfig`